include/clang/AST/Type.h

```diff
@@ -3281,9 +3286,18 @@ class FunctionProtoType : public FunctionType, public llvm::FoldingSetNode {
   /// Whether this function has a trailing return type.
   unsigned HasTrailingReturn : 1;
 
+  /// Whether this function has bounds information for parameters.
+  unsigned HasParamBounds : 1;
```

Comment:
We can see one without other.  I've added a comment explain the invariant.  We rely on ReturnBounds to be null to indicate that there is no return bounds for a function.